sys/mips/malta/malta_mp.c

```diff
@@ -244,6 +244,10 @@ platform_start_ap(int cpuid)
 	 * reg = 0x80000000;
 	 * mttc0(2, 3, reg);
 	 */
+#if defined(CPU_QEMU_MALTA)
```

Comment:
I wonder if this should be on for malta always and not just for QEMU?  The comment implies to me that this bit is incomplete anyway.  @bukinr added this file a few years ago with this comment and I think it's only been tested with QEMU anyway.  I would probably axe the block comment along with the #ifdef.  You could perhaps have a single line comment of `/* Set entry point to _start. */` (though I'm surprised we aren't using `mpentry` from mpboot.S which would would avoid having to check the platform_id in _start?).  I think I also prefer `_start` to `locore` for the entry point name.

Reply:
* We can't jump directly to `mpentry`, it seems.  I'm not sure which bit(s) of `_start` are expected to run, but we could likely copy just those to a chunk in `asm_malta.S` if you'd prefer that we avoid re-running `locore`.

* I'm happy to repoint this at `_start` instead.

Reply:
Linux always does it for their equivalent version of SMP[1] (they have a whole myriad of other configurations you can choose on weird and wonderful platforms, but if you follow the Kconfig back this is their "MIPS MT SMP support (1 TC on each available VPE)" option). Though they also explicitly set up `sp` and `gp` in the TC so I'm not sure they're the best example...

[1] https://github.com/torvalds/linux/blob/2f4c53349961c8ca480193e47da4d44fdb8335a8/arch/mips/kernel/smp-mt.c#L142-L176

Reply:
I don't think it's worth it to try to come up with a malta version of mpentry.  I would prefer `_start` and removing the #ifdef though.